worker.py

```diff
@@ -890,6 +899,57 @@ def bag_full(self):
                     self.logger.warning("Failed to apply incubator {} on {}, code: {}".format(
                         inc.get('id', 0), egg.get('id', 0), ret))
 
+    async def handle_captcha(self, responses):
+        if not config.CAPTCHA_KEY:
+            self.logger.error("CAPTCHA Key is not set! this should not happen")
+            raise CaptchaException
+
+        if self.num_captchas >= config.CAPTCHA_RETRIES:
+            self.logger.error("User {} encountered a CAPTCHA to often, removing".format(self.username))
+            raise CaptchaException
+
+        self.num_captchas += 1
+
+        url = responses.get('CHECK_CHALLENGE', {}).get('challenge_url', ' ')
+        site_key = '6LeeTScTAAAAADqvhqVMhPpr_vB9D364Ia-1dSgK'
+        s = requests.Session()
+        try:
+            requrl = "http://2captcha.com/in.php?key={}&method=userrecaptcha&googlekey={}&pageurl={}"
+            response = s.post(requrl.format(config.CAPTCHA_KEY, site_key, url)).text
+            if not response.startswith('OK|'):
+                self.logger.error("Failed to submit CAPTCHA for solving: {}".format(response))
+                raise CaptchaSolveException
+
+            captcha_id = str(response.split('|')[1])
+
+            # Get the response, retry every 5 seconds if its not ready
+            while True:
+                recaptcha_response = s.get("http://2captcha.com/res.php?key={}&action=get&id={}"
+                        .format(config.CAPTCHA_KEY, captcha_id)).text
+                if not 'CAPCHA_NOT_READY' in recaptcha_response:
+                    break
+                sleep(5)
+
+            if not recaptcha_response.startswith('OK|'):
+                self.logger.error("Failed to get CAPTCHA response: {}".format(recaptcha_response))
+                raise CaptchaSolveException
+            token = str(recaptcha_response.split('|')[1])
+        except Exception as e:
+            self.logger.error('Got an error with getting token from recaptcha. ' +
+                    'Check your Api Key and account balance.')
+            self.logger.error(e)
+            raise CaptchaSolveException
```

Comment:
```python
except Exception as e:
    self.logger.exception('Got an error with getting token from 2Captcha.'
                          ' Check your API Key and account balance.')
    raise CaptchaSolveException from e
```
`self.logger.exception` automatically includes the trace. Also, the `+` shouldn't be necessary for combining strings within parentheses, and I think you meant to say "2Captcha" instead of "recaptcha".